psi4/driver/procrouting/proc.py

```diff
@@ -3061,6 +3108,271 @@ def run_adc(name, **kwargs):
     return core.adc(ref_wfn)
 
 
+def run_adcc(name, **kwargs):
+    """Prepare and run an ADC calculation in adcc, interpret the result and return
+    as a wavefunction.
+
+    """
+    # TODO Maybe it would improve readability if this function was spilt
+    #      up and the whole thing went to a separate file (like for sapt,
+    #      interface_cfour.py, ...
+
+    try:
+        import adcc
+        from adcc.backends import InvalidReference
+    except ModuleNotFoundError:
+        raise ValidationError("adcc extras qc_module not available. Try installing "
+            "via 'pip install adcc' or 'conda install -c adcc adcc'.")
+
+
+    if core.get_option('ADC', 'REFERENCE') not in ["RHF", "UHF"]:
+        raise ValidationError('ADC requires reference RHF or UHF')
+
+    # Bypass the scf call if a reference wavefunction is given
+    ref_wfn = kwargs.pop('ref_wfn', None)
+    if ref_wfn is None:
+        ref_wfn = scf_helper(name, use_c1=True, **kwargs)
+
+    # Start timer
+    do_timer = kwargs.pop("do_timer", True)
+    if do_timer:
+        core.tstart()
+
+    #
+    # Build kwargs for adcc
+    #
+    kwargs.pop("molecule", None)
+
+    if ref_wfn.frzcpi()[0] > 0:
+        kwargs["frozen_core"] = ref_wfn.frzcpi()[0]
+    if ref_wfn.frzvpi()[0] > 0:
+        kwargs["frozen_virtual"] = ref_wfn.frzvpi()[0]
+    if core.get_option("ADC", "NUM_CORE_ORBITALS"):
+        kwargs["core_orbitals"] = core.get_option("ADC", "NUM_CORE_ORBITALS")
```

Comment:
All: should we try to make this defaultable from quantities in wfn and existing electron spaces options?

Reply:
Because, as discussed Wed., I'm currently changing a lot of the existing electron space logic...

What's the difference between frozen core and core orbitals?

Reply:
> What's the difference between frozen core and core orbitals?

Core orbitals is specific to ADC variants with the core-valence separation approximation applied (used for X-ray spectroscopy for example) and defines how many orbitals to put in this "special" core space. On top of that you may always do classic frozen core as well, so you need two things.

Reply:
Okay, a "wavefunciton" specific orbital space. (I've heard of CVS, but every presentation of linear response I've seen has gone over my head.)

I cannot imagine a sensible default from existing wavefunction quantities/spaces.

For what it's worth, a PR I'm working on will allow wavefunctions to define custom orbital spaces to override the current frozen core / occupied / virtual / frozen virtual which is imposed on all wavefunctions, fitting or not. But that space still needs to be filled somehow, and the PR won't give any new ways to do that.

Reply:
If you want to think of it like so, yes. 

Core orbitals is not really defaultable, because it depends on what you want to compute. See https://github.com/mfherbst/psi4/blob/integrate-adcc/doc/sphinxman/source/adc.rst under **Using the core-valence separation**.

By the way: CVS is not just a concept of linear response or propagator theory ... people in CI use it, too ... they just use different names for it (like restricted channel or something similar). It just means that you neglect some couplings in the matrix describing your states and thus are able to focus on state, which would otherwise come much higher in energy, embedded in the valence continuum.